pkg/apis/application/v1alpha1/types.go

```diff
@@ -2398,6 +2398,17 @@ func (source *ApplicationSource) ExplicitType() (*ApplicationSourceType, error)
 
 // Equals compares two instances of ApplicationDestination and return true if instances are equal.
 func (dest ApplicationDestination) Equals(other ApplicationDestination) bool {
+	// ignore destination cluster name and isServerInferred fields during comparison
```

Comment:
Do you mean to say that "ignore destination cluster URL" because we are setting `dest.Server = ""`?